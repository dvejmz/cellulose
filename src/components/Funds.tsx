import React from 'react';

interface FundsProps {
  amount: number;
  currency: string;
}

const Funds: React.FC<FundsProps> = (props: FundsProps) => {
  return (
    <div className="resources__funds">
      Funds: <span className="resources__funds-currency">{props.currency}</span><span className="resources__funds-amount">{props.amount}</span>
    </div>
  );
};

export default Funds;
