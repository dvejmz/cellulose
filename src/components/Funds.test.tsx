import { shallow } from 'enzyme';
import React from 'react';
import Funds from './Funds';

describe('<Funds />', () => {
  let funds: any;

  beforeEach(() => {
    funds = shallow(<Funds amount={100} currency={'£'} />);
  });

  it('shows amount', () => (
    expect(funds.find('.resources__funds-amount').text()).toEqual('100')
  ));

  it('shows currency', () => (
    expect(funds.find('.resources__funds-currency').text()).toEqual('£')
  ));
});
