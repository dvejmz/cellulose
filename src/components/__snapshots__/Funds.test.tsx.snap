// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Funds /> renders 1`] = `
<div
  className="funds"
  data-test-id="funds"
>
  <strong>
    Funds:
  </strong>
   
  <span
    className="funds__currency"
  >
    Â£
  </span>
  <span
    className="funds__amount"
  >
    100.00
  </span>
</div>
`;
